tests/spread/integration/coreutils/task.yaml

```
@@ -1,78 +1,2 @@
-summary: Integration tests for coreutils
-
-prepare: |
-  # Deb arch to GOARCH.
-  arch="$(dpkg --print-architecture | sed -e 's/armhf/arm/g' -e 's/ppc64el/ppc64le/g')"
-  # Install yq.
-  wget https://github.com/mikefarah/yq/releases/latest/download/yq_linux_${arch} \
-    -O /usr/bin/yq
-  chmod +x /usr/bin/yq
-  yq --version
-
-execute: |
-  # Run smoke tests.
-  ./smoke.sh
-
-  # Run single-binary slices tests
-  # Forward port from 24.04
-
-  # test env
-  rootfs_env="$(install-slices coreutils_env)"
-  chroot "$rootfs_env" env --version
-
-  # test expr binary
-  rootfs_expr="$(install-slices coreutils_expr)"
-  chroot "$rootfs_expr" expr --version
-
-  # test mkdir binary
-  rootfs_mkdir="$(install-slices coreutils_mkdir)"
-  chroot "$rootfs_mkdir" mkdir test_dir
-  test -d "$rootfs_mkdir/test_dir"
-
-  # test echo binary
-  rootfs_echo="$(install-slices coreutils_echo)"
-  chroot "$rootfs_echo" echo "Hello, World!"
-
-  # test ln binary
-  rootfs_ln="$(install-slices coreutils_ln-utility)"
-  touch "$rootfs_ln/test_file"
-  chroot "$rootfs_ln" ln -s test_file test_link
-  test -L "$rootfs_ln/test_link"
-
-  # test rm binary
-  rootfs_rm="$(install-slices coreutils_rm-utility)"
-  touch "$rootfs_rm/test_file"
-  chroot "$rootfs_rm" rm test_file
-  test ! -e "$rootfs_rm/test_file"
-
-  # test readlink binary
-  rootfs_readlink="$(install-slices coreutils_readlink)"
-  touch "$rootfs_readlink/test_file"
-  ln -s test_file "$rootfs_readlink/test_link"
-  chroot "$rootfs_readlink" readlink test_link | grep "test_file"
-
-  # test sort binary
-  rootfs_sort="$(install-slices coreutils_sort)"
-  echo -e "banana\napple\ncherry" > "$rootfs_sort/test_file"
-  chroot "$rootfs_sort" sort test_file > sorted_output
-  # NOTE: sorted output has a trailing newline. replaced by hyphen to show better
-  test "$(cat sorted_output | tr '\n' '-')" = "apple-banana-cherry-"
-
-  # test dirname binary
-  rootfs_dirname="$(install-slices coreutils_dirname)"
-  mkdir -p "$rootfs_dirname/foo/bar"
-  touch "$rootfs_dirname/foo/bar/baz.txt"
-  test "$(chroot "$rootfs_dirname" dirname /foo/bar/baz.txt)" = "/foo/bar"
-  test "$(chroot "$rootfs_dirname" dirname /foo/bar/)" = "/foo"
-
-  # test touch binary
-  rootfs_touch="$(install-slices coreutils_touch)"
-  chroot "$rootfs_touch" touch test_file
-  test -e "$rootfs_touch/test_file"
-
-  # test printf binary
-  rootfs_printf="$(install-slices coreutils_printf)"
-  test "$(chroot "$rootfs_printf" printf "hello\n" )" = "hello"
-  test "$(chroot "$rootfs_printf" printf "hello-%s\n" "world")" = "hello-world"
-  test "$(chroot "$rootfs_printf" printf "number: %d\n" 42)" = "number: 42"
-  test "$(chroot "$rootfs_printf" printf "float: %.2f\n" 3.14159)" = "float: 3.14"
+summary: Integration tests for apt
+execute: ./run.sh --spread
```

IMO this would be easier to follow using variants.

```suggestion
environment:
  SCRIPT/smoke: test_smoke.sh
  SCRIPT/subslices: test_subslices.sh
execute: "./${SCRIPT}" --spread
```

---

Agreed. To briefly restate my reply there, I'm not going to block these PRs on that as based on your response it sounds like something that needs more thought.